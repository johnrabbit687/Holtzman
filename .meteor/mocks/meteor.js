

exports.Meteor = {
  loginWithPassword: jest.fn(),
  loginWithFacebook: jest.fn(),
  methods: jest.fn(),
  call: jest.fn(),
  subscribe: jest.fn(),
  publish: jest.fn(),
  isServer: false,
  settings: {
    public: {
      heighliner: "/graphql",
    },
  }
};
