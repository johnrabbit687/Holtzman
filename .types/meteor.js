declare module AccountsGlobal {
  declare function forgotPassword(options: { email: string, callback?: Function }): void;
}

declare var Accounts: $Exports<'AccountsGlobal'>;
