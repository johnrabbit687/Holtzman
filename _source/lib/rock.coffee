###

  Rock.name

  @example get a string with the name of Rock

    console.log Rock.name

###
Rock.name = "Rock"

###

  Rock.isAlive

  @example See if Rock is online

    if (Rock.isAlive())
      debug "its alive!"

###
Rock.isAlive = ->

  serverWatch.isAlive Rock.name


###

  Rock.isAlive.onChange

  @example Calls the callback function immediately and then everytime Rock's
    status changes with a single boolean param of isAlive

    Rock.isAlive.onChange (isAlive) ->
      console.log isAlive

###
Rock.isAlive.onChange = (callback) ->

  serverWatch.onChange Rock.name, callback
