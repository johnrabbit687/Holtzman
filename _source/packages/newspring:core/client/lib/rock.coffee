Template.registerHelper "Rock", Rock

Meteor.call("rock-url", (err, name) ->
  Rock.baseURL = name
)
