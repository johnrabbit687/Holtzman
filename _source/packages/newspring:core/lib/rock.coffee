###

  Rock.name

  @example get a string with the name of Rock

    console.log Rock.name

###
Rock.name = "Rock"

###

  Rock.isAlive

  @example See if Rock is online

    if (Rock.isAlive())
      debug "its alive!"

###
Rock.isAlive = ->

  serverWatch.isAlive Rock.name
