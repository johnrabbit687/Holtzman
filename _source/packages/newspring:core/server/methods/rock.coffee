Meteor.methods
  "rock-url": ->
    return Rock.baseURL
