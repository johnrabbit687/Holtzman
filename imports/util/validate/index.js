import { creditCVV, creditCard, creditExpiry } from "./credit-card";
import { zipCode, locationBasedZipCode } from "./zipcode";

const Validate = {};

// XXX deprecated loading method
Validate.addValidator = (name, handler) => {
  if (Validate[name]) {
    throw new Error("Validator assigned", `Validate function ${name} is already registered`);
  }

  if (!handler || typeof handler !== "function") {
    throw new Error("Validator TypeError", `Validator ${name} requires a function`);
  }

  Validate[name] = handler;
  return { [name]: handler };
};

Validate.addValidator("isCCV", creditCVV);
Validate.addValidator("isCreditCard", creditCard);
Validate.addValidator("isExpiry", creditExpiry);
Validate.addValidator("isZipCode", zipCode);
Validate.addValidator("isLocationBasedZipCode", locationBasedZipCode);

export default Validate;
