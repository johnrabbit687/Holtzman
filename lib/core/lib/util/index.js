"use strict";

exports.__esModule = true;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _error = require("../error");

var _error2 = _interopRequireDefault(_error);

var _inherit = require("./inherit");

var _encode = require("./encode");

var _guid = require("./guid");

var _xml = require("./xml");

var _xml2 = _interopRequireDefault(_xml);

var Util = {};

Util.addUtil = function (name, handler) {

  if (Util[name]) {
    throw new _error2["default"]("Util assigned", "Util function " + name + " is already registered");
  }

  if (!handler || typeof handler != "function") {
    throw new _error2["default"]("Util TypeError", "Util " + name + " requires a function");
  }

  Util[name] = handler;
  return;
};

Util.addUtil("inherit", _inherit.inherit);

Util.addUtil("base64Encode", _encode.base64Encode);

Util.addUtil("makeNewGuid", _guid.makeNewGuid);

Util.addUtil("parseXML", _xml2["default"].parse);

exports["default"] = Util;
module.exports = exports["default"];