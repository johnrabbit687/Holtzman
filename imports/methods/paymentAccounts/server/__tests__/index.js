
import { Meteor } from "meteor/meteor";
import { api } from "../../../../util/rock";

import remove from "../";

const mockedData = {
  ReferenceNumber: "TEST",
  FinancialGatewayId: 3,
  Id: 1,
};

jest.mock("../../../../util/rock", () => ({
  api: {
    _: { give: { gateway: { id: 3 } } },
    get: { sync: jest.fn(() => mockedData) },
    delete: { sync: jest.fn(() => true) },
  }
}));

it("correctly adds the function to Meteor.methods under `PaymentAccounts.remove`", () => {
  expect(Meteor.methods).toHaveBeenCalledTimes(1);
  expect(Meteor.methods.mock.calls[0][0]).toEqual({ "PaymentAccounts.remove": remove });
});

it("correctly removes an account", () => {
  const id = 1;
  const mockedWrappedAsync = jest.fn(() => jest.fn(details => {
    expect(details).toEqual(mockedData.ReferenceNumber);
    return true;
  }));
  const originalAsync = Meteor.wrapAsync;
  Meteor.wrapAsync = mockedWrappedAsync;

  const success = remove(id);
  expect(api.get.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(api.delete.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(mockedWrappedAsync).toHaveBeenCalled();
  expect(success).toBe(true);

  Meteor.wrapAsync = originalAsync;
});

it("throws a helpful error if account can't be removed", () => {
  const id = 1;

  // Mocks
  const mockedDelete = jest.fn(() => { status: "ERROR" });
  const originalDelete = api.delete.sync;
  api.delete.sync = mockedDelete;

  const mockedWrappedAsync = jest.fn(() => jest.fn(details => {
    expect(details).toEqual(mockedData.ReferenceNumber);
    return true;
  }));
  const originalAsync = Meteor.wrapAsync;
  Meteor.wrapAsync = mockedWrappedAsync;

  let succes;
  try {
    remove(id);
  } catch (e) {
    expect(e).toMatchSnapshot();
  }
  expect(api.get.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(api.delete.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(mockedWrappedAsync).not.toHaveBeenCalled();

  // repair mocks
  Meteor.wrapAsync = originalAsync;
  api.delete.sync = originalDelete;
});


it("throws a helpful error if account lookup fails", () => {
  const id = 1;

  // Mocks
  const mockedDelete = jest.fn(() => undefined);
  const originalDelete = api.delete.sync;
  api.delete.sync = mockedDelete;

  const mockedWrappedAsync = jest.fn(() => jest.fn(details => {
    expect(details).toEqual(mockedData.ReferenceNumber);
    return true;
  }));
  const originalAsync = Meteor.wrapAsync;
  Meteor.wrapAsync = mockedWrappedAsync;

  let succes;
  try {
    remove(id);
  } catch (e) {
    expect(e).toMatchSnapshot();
  }
  expect(api.get.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(api.delete.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(mockedWrappedAsync).not.toHaveBeenCalled();

  // repair mocks
  Meteor.wrapAsync = originalAsync;
  api.delete.sync = originalDelete;
});

it("does not try to remove an account if not containing a ReferenceNumber", () => {
  const id = 1;

  // mocks
  const mockedWrappedAsync = jest.fn(() => jest.fn(details => {
    expect(details).toEqual(mockedData.ReferenceNumber);
    return true;
  }));
  const originalAsync = Meteor.wrapAsync;
  Meteor.wrapAsync = mockedWrappedAsync;

  const originalGet = api.get.sync;
  const mockedGet = jest.fn(() => ({ status: "Error" }));
  api.get.sync = mockedGet;

  const success = remove(id);
  expect(api.get.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(api.delete.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(mockedWrappedAsync).not.toHaveBeenCalled();
  expect(success).toBe(true);

  Meteor.wrapAsync = originalAsync;
  api.get.sync = originalGet;
});

it("does not try to remove an account if id on the account does not match the gateway", () => {
  const id = 1;

  // mocks
  const mockedWrappedAsync = jest.fn(() => jest.fn(details => {
    expect(details).toEqual(mockedData.ReferenceNumber);
    return true;
  }));
  const originalAsync = Meteor.wrapAsync;
  Meteor.wrapAsync = mockedWrappedAsync;

  const originalGet = api.get.sync;
  const mockedResponse = {
    ...mockedData,
    ...{ FinancialGatewayId: 2 },
  };

  const mockedGet = jest.fn(() => mockedResponse);
  api.get.sync = mockedGet;

  const success = remove(id);
  expect(api.get.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(api.delete.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(mockedWrappedAsync).not.toHaveBeenCalled();
  expect(success).toBe(true);

  Meteor.wrapAsync = originalAsync;
  api.get.sync = originalGet;
});

it("does not try to remove an account if the gateway id does not match the id on the account", () => {
  const id = 1;

  // mocks
  const mockedWrappedAsync = jest.fn(() => jest.fn(details => {
    expect(details).toEqual(mockedData.ReferenceNumber);
    return true;
  }));
  const originalAsync = Meteor.wrapAsync;
  Meteor.wrapAsync = mockedWrappedAsync;

  const originalGatewayId = api._.give.gateway.id;
  api._.give.gateway.id = 4;

  const success = remove(id);
  expect(api.get.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(api.delete.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(mockedWrappedAsync).not.toHaveBeenCalled();
  expect(success).toBe(true);

  Meteor.wrapAsync = originalAsync;
  api._.give.gateway.id = originalGatewayId;
});

it("correctly throws if call to NMI fails and includes messages", () => {
  const id = 1;
  const mockedWrappedAsync = jest.fn(() => jest.fn(details => {
    expect(details).toEqual(mockedData.ReferenceNumber);
    throw new Error("NMI Transactaion Failure");
    return true;
  }));
  const originalAsync = Meteor.wrapAsync;
  Meteor.wrapAsync = mockedWrappedAsync;

  let success;
  try {
    success = remove(id);
  } catch (e) {
    expect(e).toMatchSnapshot();
    expect(e.message).toContain("NMI Transactaion Failure");
  }
  expect(api.get.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(api.delete.sync.mock.calls[0][0]).toBe(`FinancialPersonSavedAccounts/${id}`);
  expect(mockedWrappedAsync).toHaveBeenCalled();
  expect(success).toBeFalsy();

  Meteor.wrapAsync = originalAsync;
});
