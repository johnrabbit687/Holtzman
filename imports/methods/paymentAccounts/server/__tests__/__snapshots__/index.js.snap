exports[`test correctly throws if call to NMI fails and includes messages 1`] = `[Error: NMI Transactaion Failure]`;

exports[`test throws a helpful error if account can't be removed 1`] = `
[Error: 
      Something when wrong when removing your payment account, please try again
    ]
`;

exports[`test throws a helpful error if account lookup fails 1`] = `
[Error: 
      Something when wrong when removing your payment account, please try again
    ]
`;
